de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/ABeanStructuralElementInstance.java

```diff
@@ -219,20 +219,49 @@ public void setChildren(List<ABeanStructuralElementInstance> newBeanSeis) {
 	}
 	
 	/**
-	 * Shadows the original function, but makes the list modifiable
-	 * so it can be used by JAXB
+	 * Create a list of BeanStructuralElementInstanceReference from a list of seis
+	 * @param seis list of seis
+	 * @return list of BeanStructuralElementInstanceReference
 	 */
-	public List<ABeanStructuralElementInstance> getJaxbChildren() {
-		return new ArrayList<ABeanStructuralElementInstance>(getChildren());
+	private List<BeanStructuralElementInstanceReference> getReferenceList(List<StructuralElementInstance> seis) {
+		ArrayList<BeanStructuralElementInstanceReference> children = new ArrayList<BeanStructuralElementInstanceReference>();
+		
+		for (StructuralElementInstance sei : seis) {
+			children.add(new BeanStructuralElementInstanceReference(sei));
+		}
+		
+		return children;
+	}
+	
+	/**
+	 * Update list of seis from a list of BeanStructuralElementInstanceReference
+	 * @param currentSeis list of seis
+	 * @param newBeanSeis list of BeanStructuralElementInstanceReference
+	 */
+	private void setReferenceList(List<StructuralElementInstance> currentSeis, List<BeanStructuralElementInstanceReference> newBeanSeis) {
+		List<StructuralElementInstance> newChildren = new ArrayList<StructuralElementInstance>();
+		
+		for (BeanStructuralElementInstanceReference beanSei : newBeanSeis) {
+			StructuralElementInstance sei = beanSei.getStructuralElementInstance();
+			newChildren.add(sei);
+		}
+		
+		currentSeis.clear();
+		currentSeis.addAll(newChildren);
+	}
+	
+	public List<BeanStructuralElementInstanceReference> getJaxbChildren() {
+		return getReferenceList(sei.getChildren());
 	}
 	
-	//CHECKSTYLE:OFF: MagicNumber
 	@XmlElement(name = "children")
+	//CHECKSTYLE:OFF: MagicNumber
 	@ApiModelProperty(name = "children", required = true,
 		value = "List of the child beans",
 		position = 3)

```

Comment:
Why do we need this magic number here?

Reply:
Because of the `position = 3`. Without the editor shows a magic number warning. (tested it right now in my IDE to be sure)

Reply:
The alternative would be to use constants

Reply:
Also before there were only two CHECKSTYLE comments for the whole file, I now moved them around the problematic annotations to improve code quality. Maybe thats what confused you

Reply:
Yep, I know that checkstyle complaints about the magic numbers... That's what it is for :D My question was more like can't we use Constants here?

Reply:
Sure I will change it then